src/main/java/htsjdk/samtools/util/IntervalList.java

```diff
@@ -869,4 +844,68 @@ public int hashCode() {
         result = 31 * result + intervals.hashCode();
         return result;
     }
+
+    public static class IntervalMerger implements Iterable<Interval> {
+
+        Iterator<Interval> inputIntervals;
+        Interval current = null;
+        final boolean combineAbuttingIntervals;
+        final boolean enforceSameStrands;
+        final boolean concatenateNames;
+        final List<Interval> toBeMerged = new ArrayList<>();
+
+        public IntervalMerger(Iterator<Interval> intervals, final boolean combineAbuttingIntervals, final boolean enforceSameStrand, final boolean concatenateNames) {
+            this.inputIntervals = intervals;
+
+            this.combineAbuttingIntervals = combineAbuttingIntervals;
+            this.enforceSameStrands = enforceSameStrand;
+            this.concatenateNames = concatenateNames;
+        }
+
+        @Override
+        public Iterator<Interval> iterator() {
```

Comment:
if this remains iterable it needs javadoc warning that it's not actually repeatable.

Reply:
ok. moving to iterator.